src/python/pants/backend/python/subsystems/pex_build_util.py

```diff
@@ -30,6 +34,86 @@
 from pants.util.contextutil import temporary_file
 
 
+# This bootstrapping technique for lazily loading requirements comes from the pex project test case:
+# https://github.com/pantsbuild/pex/commit/66aa5ef69e8b65004a548b44edc0b81dfa8ec736#diff-b5b57c9d4fc663f59ecaedb851c13bd8R23
+_IPEX_PREAMBLE = """\
+import json
+import os
+import sys
+
+from pex import resolver, vendor
+from pex.common import open_zip
+from pex.fetcher import Fetcher, PyPIFetcher
+from pex.interpreter import PythonInterpreter
+from pex.pex_builder import BOOTSTRAP_DIR, PEXBuilder
```

Comment:
Please take a thorough pass through this code to remove all the hackery. pex.pex_builder.BOOTSTRAP_DIR, pex.third_party, pex.vendor, CacheHelper, ENV - these all scream hack when used from outside pex. The PEX API is just PexBuilder and resolver and any types they directly expose.

Reply:
Moved to a separate literal file `_ptex_launcher.py` to allow lint to work its magic!